CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/parameters/types/SingleChoiceParameterType.java

```diff
@@ -128,9 +128,22 @@
      * parameter.
      */
     public static void setOptions(final PluginParameter<SingleChoiceParameterValue> parameter, final List<String> options) {
+        //Change only if the options are changed.
+        if (optionsChanged(parameter, options)) {
         final SingleChoiceParameterValue parameterValue = parameter.getParameterValue();
+
+            //Clear the existing selection
+            parameter.setObjectValue(null);
+
         parameterValue.setOptions(options);
         parameter.setProperty(CHOICES, new Object());
+        }
+    }
+
+    private static boolean optionsChanged(PluginParameter<SingleChoiceParameterValue> parameter, List<String> options) {
+        final SingleChoiceParameterValue parameterValue = parameter.getParameterValue();
+        boolean bol = !options.equals(parameterValue.getOptions());
+        return bol;

```

Comment:
these lines can be reduced to `return !options.equals(parameter.getParameterValue().getOptions())`

Reply:
Yeah already fixed it but didn't push that file. Added that for testing.